### PR TITLE
DDST-136: Create destination plugin for creating entities with entity revisions

### PR DESCRIPTION
**Pull request by chrismacdonaldw**

### CODE REVIEW

File: src/Plugin/migrate/destination/DgiEntity.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ * Provides a custom destination plugin for generating entity revisions.
+ *
+ * This class extends the EntityContentBase class to provide a custom
+ * destination plugin for Drupal migrations. It allows for the creation of
+ * revisions during the migration process, and maintains revision history.
+ *
+ * @see \Drupal\migrate\Plugin\migrate\destination\EntityContentBase
+ *
+ * @MigrateDestination(
+ *   id = "dgi_node"
```
Comment (adam-vessey) [inline src/Plugin/migrate/destination/DgiEntity.php:23]: Might be more flexible to accept the "entity type" as plugin configuration... `dgi_revisioned_entity`, or something of the like? Sure, we might anticipate only using for "node" at present, but could be similarly usable with other revisionable entities, right?

File: src/Plugin/migrate/destination/DgiEntity.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The content entity.
+   *
+   * @return string
+   *   The generated revision message.
+   */
+  protected function generateRevisionMessage(ContentEntityInterface $entity) {
+    $entity_id = $entity->id();
+
+    return sprintf('Migration %s generated new revision for NID %d', $this->migrationId, $entity_id);
```
Comment (adam-vessey) [inline src/Plugin/migrate/destination/DgiEntity.php:97]: Not sure that including the entity ID here makes sense, given this is in the generation of a message explicitly for it.

File: src/Plugin/migrate/destination/DgiRevisionedEntity.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+ * )
+ */
+class DgiRevisionedEntity extends EntityContentBase {
+
+  /**
+   * The entity type.
+   *
+   * @var string
+   */
+  public static string $entityType;
```
Comment (adam-vessey) [inline src/Plugin/migrate/destination/DgiRevisionedEntity.php:33]: Should probably no longer be static. Could lead to confusion if ever instantiating multiple of these with different types in a single PHP process, as the type would be whichever was instantiated last.

```suggestion
  public string $entityType;
```

File: src/Plugin/migrate/destination/DgiRevisionedEntity.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+   */
+  protected string $migrationId;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition, MigrationInterface $migration = NULL) {
+    static::$entityType = $configuration['entity_type'] ?? 'node';
+    $instance = parent::create($container, $configuration, 'entity:' . static::$entityType, $plugin_definition, $migration);
+    $instance->migrationId = $migration->id();
```
Comment (adam-vessey) [inline src/Plugin/migrate/destination/DgiRevisionedEntity.php:48]:
```suggestion
    $entity_type = $configuration['entity_type'] ?? 'node';
    $instance = parent::create($container, $configuration, 'entity:' . $entity_type, $plugin_definition, $migration);
    $instance->entityType = $entity_type;
    $instance->migrationId = $migration->id();
```

File: src/Plugin/migrate/destination/DgiRevisionedEntity.php (hunk 3 above)
Comment (adam-vessey) [reply src/Plugin/migrate/destination/DgiRevisionedEntity.php:33]: Though, on second though, this value isn't really used? Could we just drop this variable entirely?

**Review by nchiasson-dgi — APPROVED** [review]

**Pull request merged**